### PR TITLE
arch: avoid mixing types

### PR DESCRIPTION
**Pull request by DeHess**
Fixed some occurrences of mixing up types in arch

This PR is part of the enhancement issue https://github.com/zephyrproject-rtos/zephyr/issues/48002 which port the coding guideline fixes done by BUGSENG on the https://github.com/zephyrproject-rtos/zephyr/tree/v2.7-auditable-branch back to main

The commit in this PR is a subset of the original auditable-branch commit:
https://github.com/zephyrproject-rtos/zephyr/commit/d03fa8d8e9d90f104e5dadafea0f773a67862659

### CODE REVIEW

File: arch/arm/core/cortex_m/irq_manage.c [hunk 1]
```diff
@@ -88,7 +88,7 @@ void z_arm_irq_priority_set(unsigned int irq, unsigned int prio, uint32_t flags)
 	__ASSERT(prio <= (BIT(NUM_IRQ_PRIO_BITS) - 1),
 		 "invalid priority %d for %d irq! values must be less than %lu\n",
 		 prio - _IRQ_PRIO_OFFSET, irq,
-		 BIT(NUM_IRQ_PRIO_BITS) - (_IRQ_PRIO_OFFSET));
+		 (unsigned long)BIT(NUM_IRQ_PRIO_BITS) - (_IRQ_PRIO_OFFSET));
```
Comment (andyross) [inline arch/arm/core/cortex_m/irq_manage.c:91]: Pretty sure the bug here is in the printf string.  The argument should be "%u" and not "%lu", the NVIC priorities are limited to a single word mask no?

File: arch/x86/core/fatal.c [hunk 2]
```diff
@@ -52,7 +52,7 @@ bool z_x86_check_stack_bounds(uintptr_t addr, size_t size, uint16_t cs)
 	if (_current == NULL || arch_is_in_isr()) {
 		/* We were servicing an interrupt or in early boot environment
 		 * and are supposed to be on the interrupt stack */
-		int cpu_id;
+		uint8_t cpu_id;
```
Comment (andyross) [inline arch/x86/core/fatal.c:55]: Again confused.  It's true that the "id" field of struct _cpu is a byte, but that's an implementation detail.   Code working with it (which is rare, and limited to low level drivers like this) wants to be using a machine word for portability and simplicity.

File: arch/x86/core/fatal.c [hunk 3]
```diff
@@ -471,29 +469,29 @@ void z_x86_page_fault_handler(z_arch_esf_t *esf)
 	dump_page_fault(esf);
 #endif
 #ifdef CONFIG_THREAD_STACK_INFO
-	if (z_x86_check_stack_bounds(esf_get_sp(esf), 0, esf->cs)) {
-		z_x86_fatal_error(K_ERR_STACK_CHK_FAIL, esf);
+	if (z_x86_check_stack_bounds(esf_get_sp(esf), 0, (uint16_t)esf->cs)) {
+		z_x86_fatal_error((unsigned int)K_ERR_STACK_CHK_FAIL, esf);
```
Comment (andyross) [inline arch/x86/core/fatal.c:473]: What's up here?  If the defined error enumerants don't work to pass to the fatal error handler, we have an API bug.  A cast is the wrong solution.

**Review by andyross — COMMENTED** [review]
Hm... there's a lot of arguable stuff in this PR, and the arch layers are subtle and error-prone, with a limited population of people who can puzzle out new interactions.

Basically this just creeps me out a bit.  Can you split some of these up thematically into smaller chunks, and/or maybe provide the list of squawks as a bug to the Intel folks and let them handle the actual patching?

**Comment by MaureenHelm** [comment]
@DeHess please take a look at the twister failures. this also needs a rebase due to merge conflicts

**Pull request closed without merging**